xrpl/utils/txn_parser/utils/parser.py

```diff
@@ -0,0 +1,46 @@
+"""Helper functions for parsers."""
+
+from decimal import Decimal
+from typing import Any, Dict, List, Union
+
+from xrpl.utils.txn_parser.utils.types import (
+    AccountBalance,
+    AccountOfferChange,
+    CurrencyAmount,
+)
+
+
+def get_value(balance: Union[CurrencyAmount, Dict[str, str], str]) -> Decimal:
+    """
+    Get a currency amount's value.
+
+    Args:
+        balance: Account's balance.
+
+    Returns:
+        The currency amount's value.
+    """
+    if isinstance(balance, str):
+        return Decimal(balance)
+    return Decimal(balance["value"])
+
+
+def group_by_account(
+    account_objects: Union[List[AccountBalance], List[AccountOfferChange]],
+) -> Dict[str, Any]:
+    """
+    Groups the account objects in one list for each account.
+
+    Args:
+        account_objects: All computed objects.
+
+    Returns:
+        The grouped computed objects.
+    """
+    grouped_objects: Dict[str, Any] = {}
+    for object in account_objects:
+        account = object["account"]
+        if account not in grouped_objects:
+            grouped_objects[account] = []
+        grouped_objects[account].append(object)
```

Comment:
I just learned a better way to do this :) 

Source with more detail and other built-in ways to do this: https://stackoverflow.com/questions/12905999/how-to-create-key-or-append-an-element-to-key

```suggestion
grouped_objects.setdefault(object["account"], []).append(object)
```